Matrix.py

```diff
@@ -0,0 +1,181 @@
+"""Matrices calculator class.
+
+The matrices calculator can be used to add, sub, multiplication (by scalars or matrices) and scalar
+division. also you can compare matrices or using them as a dictionary key. the program only works
+with squared matrices.
+
+    Typical usage example:
+
+    x = Matrix(((1,2,3),(4,5,6),(6,7,8)))
+    y = Matrix.unity(3)
+    new_matrix = x * y
+    """
+
+
+class Matrix:
+    """A matrices calculator for basic math actions.
+
+        Attributes:
+            data(tuple of tuples): a Matrix- each line is a tuple in the major tuple.
+    """
+
+    def __init__(self, data):
+        if type(data) is not tuple:
+            raise ValueError("Invalid matrix- not a tuple")
+
+        for line in data:
+            if len(line) != len(data) or type(line) is not tuple:
+                raise ValueError("Invalid matrix- lines must be tuples and as long as the matrix")
+
+        self.data = data
+
+    def __iter__(self):
+        return iter(self.data)
+
+    def __get__(self, instance, owner):
+        return self.data
```

Comment:
?

Reply:
oops, tried something and forgot to delete it